app/gui2/e2e/graphNodeVisualization.spec.ts

```diff
@@ -16,3 +16,24 @@ test('node can open and load visualization', async ({ page }) => {
   // The default JSON viz data contains an object.
   await expect(locate.jsonVisualization(page)).toContainText('{')
 })
+
+test('Warnings visualization', async ({ page }) => {
+  await actions.goToGraph(page)
+
+  // Create a node, attach a warning, open the warnings-visualization.
+  await locate.addNewNodeButton(page).click()
+  const input = locate.componentBrowserInput(page).locator('input')
+  await input.fill('Warning.attach "Uh oh" 42')
+  await page.keyboard.press('Enter')
+  await expect(locate.componentBrowser(page)).not.toBeVisible()
+  await expect(locate.circularMenu(page)).toExist()
+  await locate.toggleVisualizationButton(page).click()
+  await expect(locate.anyVisualization(page)).toExist()
+  await locate.showVisualizationSelectorButton(page).click()
+  await page.getByText('Warnings').click()
```

Comment:
I would call this `.getByText('Warnings')` on the visualization locator (`locate.anyVisualization(page)` I guess) to make sure we won't confuse it with some `Warnings` panel in the future.